lib/Runtime/Library/IntlEngineInterfaceExtensionObject.cpp

```diff
@@ -1843,17 +1807,82 @@ namespace Js
             );
         }
 
+        // As of ES2015, String.prototype.localeCompare must compare canonically equivalent strings as equal
+        BEGIN_TEMP_ALLOCATOR(tempAllocator, scriptContext, _u("EntryIntl_LocaleCompare"));
+
+        const char16 *leftNormalized = nullptr;
+        charcount_t leftNormalizedLength = 0;
+        if (UnicodeText::IsNormalizedString(UnicodeText::NormalizationForm::C, left->GetSz(), left->GetLength()))
+        {
+            leftNormalized = left->GetSz();
+            leftNormalizedLength = left ->GetLength();
+        }
+        else
+        {
+            leftNormalized = left->GetNormalizedString(UnicodeText::NormalizationForm::C, tempAllocator, leftNormalizedLength);
+        }
+
+        const char16 *rightNormalized = nullptr;
+        charcount_t rightNormalizedLength = 0;
+        if (UnicodeText::IsNormalizedString(UnicodeText::NormalizationForm::C, right->GetSz(), right->GetLength()))
+        {
+            rightNormalized = right->GetSz();
+            rightNormalizedLength = right->GetLength();
+        }
+        else
+        {
+            rightNormalized = right->GetNormalizedString(UnicodeText::NormalizationForm::C, tempAllocator, rightNormalizedLength);
+        }
+
+        END_TEMP_ALLOCATOR(tempAllocator, scriptContext);
```

Comment:
This smells like a use-after-free, unless I'm misinterpreting the meaning of END_TEMP_ALLOCATOR. Should we do this after the part where we use leftNormalized and rightNormalized below?